scimath/units/compare_units.py

```diff
@@ -0,0 +1,71 @@
+""" Utilities around unit comparisons.
+"""
+import numpy as np
+
+from scimath.units.api import convert, UnitArray, UnitScalar
+from scimath.units.unit import InvalidConversion
+
+
+def unit_scalars_almost_equal(x1, x2, eps=1e-9):
```

Comment:
I think we really want `eps` to be a quantity of the same unit class as `x1` and `x2` here, and I'd suggest removing the default value and making it a required argument. It just doesn't make sense conceptually to ask whether two lengths (for example) are equal up to two decimal places, while it absolutely makes sense to ask whether two lengths are equal to within a margin of 2cm, or 20nm, or 1/8 inch, or ...

It's particularly unclear what comparing to `1e-9` would mean in the case that `x1` and `x2` are comparable, but have different actual units, and the implementation here looks as though it would give asymmetric results:

```
>>> x = UnitScalar(1.0, units="um")
>>> y = UnitScalar(1002, units="nm")
>>> unit_scalars_almost_equal(x, y, eps=1e-2)
True
>>> unit_scalars_almost_equal(y, x, eps=1e-2)
False
```

Reply:
Thanks for the valuable feedback @mdickinson . I totally agree with you on eps needing to be unitted, and for it to be a required argument. I would only suggest to allow for eps to be a float if the 2 values have the same unit. I will push an update...

Reply:
I'd go further, and only allow use of a `float` eps if both quantities being compared are actually unitless. Otherwise you're again comparing a unitful quantity with a unitless quantity, which seems like a category error to me. It would seem wrong to me if a given `unit_scalars_almost_equal` passes for some length `length1` and fails for a `length2` that represents exactly the same length, but happens to use different units.

Reply:
IOW, I'd want the check to be checking a property of the length itself, not a property of the representation of that length.

Reply:
Thanks, I see your point. At the same time, I would like to keep it simple to use. What you propose would lead to this for the usage
```
from scimath.units.testing.assertion_utils import assert_unit_scalar_almost_equal
from scimath.units.api import UnitScalar
assert_unit_scalar_almost_equal(x, y, eps=UnitScalar(1e-12, units="SOME UNIT HERE"))
```
is quite verbose, and makes it hard to write lots of tests. Let's say that my use case is to check the content of 2 serialized projects with lots of scimath quantities inside. The way would end up using this would be to do:
```
assert_unit_scalar_almost_equal(x, y, eps=UnitScalar(1e-12, units=x.units))
```
What about automatically treating a float `eps` like `UnitScalar(eps, units=x.units)` inside `unit_scalars_almost_equal`? Or change `eps` to a non-dimensional "`rtol`" which would be compared to `float(abs(x1-x2)/abs(x2))` (after unit conversion) similar to https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.allclose.html ?